docs/proposals/konnectivity-openvpn.md

```diff
@@ -0,0 +1,120 @@
+
+
+## Motivation and Background
+
+The current solution for communication between k8s control plane components and worker nodes is custom to KKP and quite complex. It relies on OpenVPN tunnels between the control plane cluster and user clusters, deployed as:
+
+* OpenVPN server pod in the control plane cluster,
+* OpenVPN client pod in user clusters,
+* Custom iptables configuration deployed as init containers in OpenVPN pods,
+* `openvpn-client` and `dnat-controller` sidecar containers in several control plane components.
+
+This complex solution may be prone to errors especially in large scale deployments. In case of issues, it is not easy to debug them without deep knowledge of this custom setup. With Konnectivity, all these components can be replaced with just Konnectivity Server in the control plane cluster and Konnectivity Agent(s) in the user cluster.
+
+As part of the implementation, we also propose moving the components that act as k8s API extensions (e.g. `metrics-server`) into the user cluster, which would allow us to simplify our stack even more, by getting rid of the custom `dnsConfig` in control plane pods and dns-resolver deployment in the control plane cluster.
+
+Note that Konnectivity is beta since k8s 1.18, so we need to support both OpenVPN and Konnectivity strategy for our users that use older k8s versions, or do not want to move to Konnectivity yet.
+
+
+## Implementation
+
+The implementation proposal follows [this documentation][kubernetes_konnectivity_doc]
+
+
+### Kube API Server Flags
+
+Konnectivity requires [Service Account Token Volume Projection][kubernetes_service_account] to be enabled with `system:konnectivity-server` in `api-audiences`. That means:
+
+* In case that the service account token volume projection is not yet enabled (which may be already done as part of [this issue][issue_6191]), we need to add some more kube-apiserver flags if Konnectivity is enabled.
+* In case that service account token volume projection is already enabled, we need to add `system:konnectivity-server` into `api-audiences` list.
+
+Apart from that, the API Server needs the `--egress-selector-config-file` flag pointed to the file with `EgressSelectorConfiguration`, which defines the type of connection between the API Server and the Konnectivity Server. 
+
+### Konnectivity Server in Control Plane Cluster
+
+The Konnectivity server runs in the control plane cluster and acts as a proxy between the Kube API server running in the same cluster and Konnectivity Agents that are running in the user cluster and are connecting to the Konnectivity Server’s IP address, which needs to be reachable from the user cluster (for this connection, Kubermatic uses various [Expose Strategies][k8c_expose_strategies]).
+
+Konnectivity server will be deployed as a sidecar container for the Kubernetes API Server. Connection between the two will be done by using UDS.
```

Comment:
We could mention that we will use an `emptyDir` volume to be able to use UDS between KAS and Konnectivity server.

Reply:
Pushed the fix